src/schema/posts.ts

```diff
@@ -2353,6 +2409,70 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         ),
       }));
     },
+    deleteSourcePostModeration: async (
+      _,
+      { postId }: { postId: string },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      const moderation = await ctx.con
+        .getRepository(SourcePostModeration)
+        .findOneByOrFail({ id: postId });
+
+      await ensureSourcePermissions(ctx, moderation.sourceId);
+
+      const isModerator = await isPrivilegedMember(ctx, moderation.sourceId);
+      const isAuthor = moderation.createdById === ctx.userId;
+
+      if (!isModerator && !isAuthor) {
+        throw new ForbiddenError('Access denied!');
+      }
+
+      await ctx.con.getRepository(SourcePostModeration).delete({ id: postId });
+
+      return { _: true };
+    },
+    editSourcePostModeration: async (
```

Comment:
This should go through the same process of creating. Maybe we can create a generic function that will be used by both mutations.

Reply:
I can help with this one if you are working on something.

Reply:
Yes thanks, would appreciate if you are free atm ✔️

Reply:
Agree with @sshanzel